app/src/main/java/com/example/campusguide/DirectionsActivity.kt

```diff
@@ -63,7 +63,16 @@ class DirectionsActivity : AppCompatActivity(), AdapterView.OnItemClickListener
         setContentView(R.layout.activity_directions)
 
         map = GoogleMapAdapter()
-        val initializer = GoogleMapInitializer(this, map, "directions_activity_map")
+        //TODO: Refactor GoogleMapInitializer so it has less nullable constructor properties
+        val initializer = GoogleMapInitializer(
+            this,
+            map,
+            "directions_activity_map",
+            null,
+            null,
+            null,
+            BuildingIndexSingleton.getInstance(assets)
+        )
```

Comment:
What is the reason for all of the null values in the constructor?

Reply:
Answer:
GoogleMapsInitializer has a large amount of optional parameters that determine which features are enabled. For example, you can add a polygon click listener which is necessary for popups when clicking on Concordia buildings. However, in activities where you don't want those features, you can just pass null.

The BuildingIndexSingleton is one of those optional parameters, it just happens to be the last one in the list. So I need to pass null for all the previous optional parameters.

This is clearly a big code smell, so I added a TODO which you can see a few lines higher to represent that piece of technical debt which I do not have time to address right now.